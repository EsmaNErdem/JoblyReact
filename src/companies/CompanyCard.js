import { Link } from "react-router-dom";

/**
 * Show short info about a company
 * 
 * - Rendered by CompanyList to display card for each company
 * - CompanyList ==> CompanyCard
 * 
 */

const CompanyCard = ({handle, name, description, logoUrl}) => {

    return (
        <Link to={`/companies/${handle}`} className="CompanyCard">
            <div >
                <h2 data-testid="company-name">
                    {name}
                    {/* {logoUrl && <img alt={name} src={logoUrl}/>} */}
                </h2>
                <p>{description}</p>
            </div>
        </Link>
    )
}

export default CompanyCard;