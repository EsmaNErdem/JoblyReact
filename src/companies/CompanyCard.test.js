import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="test"
            name="Test"
            description="testing 1, 2, 3..."
            logo_url=""
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="test"
            name="Test"
            description="testing 1, 2, 3..."
            logo_url=""
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
