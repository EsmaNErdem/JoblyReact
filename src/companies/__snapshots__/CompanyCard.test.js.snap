// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard"
    href="/companies/test"
  >
    <div>
      <h2
        data-testid="company-name"
      >
        Test
      </h2>
      <p>
        testing 1, 2, 3...
      </p>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard"
    href="/companies/test"
  >
    <div>
      <h2
        data-testid="company-name"
      >
        Test
      </h2>
      <p>
        testing 1, 2, 3...
      </p>
    </div>
  </a>
</DocumentFragment>
`;
